### PR TITLE
Update SIP-63: Trading Incentives

### PR DESCRIPTION
**Pull request by jacko125**
possible implementation of a trading incentives contract

When opening a pull request to submit a new SIP, please use the suggested template: https://github.com/Synthetixio/SIPs/blob/master/sip-X.md

We have a GitHub bot that automatically merges some PRs. It will merge yours immediately if certain criteria are met:

 - The PR edits only existing WIP PRs.
 - The build passes.
 - Your Github username or email address is listed in the 'author' header of all affected PRs, inside <triangular brackets>.
 - If matching on email address, the email address is the one publicly listed on your GitHub profile.

### CODE REVIEW

File: SIPS/sip-63.md [hunk 1]
```diff
@@ -28,21 +28,80 @@ There is currently no way to track trading fees paid on-chain, so it is not poss
...
+**Interface**
+
+```
+pragma solidity >=0.4.24;
+
+interface ITradingIncentive {
+  // Views
+  function rewards(uint periodID) external view; 
+  
+  function rewardForPeriods(uint[] periodIDs) external view;
```
Comment (justinjmoses) [inline SIPS/sip-63.md:69]: Wouldn't you also have a view so an account can see their specific rewards?

File: SIPS/sip-63.md [hunk 2]
```diff
@@ -28,21 +28,80 @@ There is currently no way to track trading fees paid on-chain, so it is not poss
...
+**Exchanger contract**
+
+Remit exchange fees generated by an account to the `Trading incentive` contract, storing the total fees for each account in the contract. 
+
+A function to allow trading incentives to be recorded against any `address` instead of the `msg.sender` of the `synthetix.exchange()` event. The `address` will need to be the one claiming the rewards from the `trading incentive` contract.    
+
 ### Test Cases
 <!--Test cases for an implementation are mandatory for SIPs but can be included with the implementation..-->
-Test cases for an implementation are mandatory for SIPs but can be included with the implementation.
+- When an exchange is made, the exchange fee in sUSD and the address the exchange `from` gets stored in the trading incentives contract and the `totalFeesInPeriod` is increased by the exchange fee amount.
```
Comment (justinjmoses) [inline SIPS/sip-63.md:91]: It's worth being explicit and adding a test case when invoked via a third party/contract, etc.

File: SIPS/sip-63.md [hunk 3]
```diff
@@ -28,21 +28,80 @@ There is currently no way to track trading fees paid on-chain, so it is not poss
...
+- When an exchange is made, if an address is passed in to record the trading incentives for, then it will store the total fees against that address.
+- When notifyRewards is called, if the SNX balance is already transferred to the contract, then it passes. The transaction will record the `totalFeesInPeriod` for the current period and reset it. It will also increment the `periodID`.
+
+**Reward Claiming**
+
+- When a user claim their rewards, their reward will be calculated as a proportion of the fees they generated against the total fees of that period and the period is marked claimed for the user.
+- When a user claim their rewards, if the claim is successful, it will record the final `totalFeesInPeriod` for the last  reward period if the period is closed, and reset their current `totalFeesInPeriod`. 
+- When a user claim their rewards, if the claim is successful, it will increase the `totalRewardsClaimed` amount which ensures the contract has enough SNX to pay the rewards (on notifyRewards it will check the contract balance less `totalRewardsClaimed`.   
+- When a user claim their rewards, if the period they provide has no fees generated, then it will fail.
```
Comment (justinjmoses) [inline SIPS/sip-63.md:102]: [np] This is better written as `Given When Then` using cucumber BDD syntax. So `Given a period that has no fees for a user, when that user clams for that period, then the transaction fails` or some such

File: SIPS/sip-63.md (hunk 1 above)
Comment (jacko125) [reply SIPS/sip-63.md:69]: Good idea, there should be account specific views

Comment (jacko125) [reply SIPS/sip-63.md]: updated

**Pull request merged**